src/misc/roctx.cc

```diff
@@ -96,4 +94,42 @@ void stringify(roctxPayloadInfo_t payloadInfo) {
   }
 
   snprintf(payloadInfo->message + offset, MAX_MESSAGE_LENGTH - offset, "}");
+}
+
+RCCL_PARAM(LogRoctx, "LOG_ROCTX", 0);
+
+roctx_scoped_range_in::roctx_scoped_range_in(const nvtxPayloadSchemaEntry_t* schema, const nvtxPayloadData_t* data, 
+                                const size_t numEntries, const char* schemaName) noexcept {
+  if (rcclParamLogRoctx()) {
+    roctxAlloc(&payloadInfo, numEntries);
+    extractPayloadInfo(schema, data, numEntries, schemaName, &payloadInfo);
+#ifdef ROCTX_ENABLE
+    roctxRangePushA(payloadInfo.message);
```

Comment:
Is it necessary to extract payload info when this feature is disabled?
```suggestion
#ifdef ROCTX_ENABLE
    roctxAlloc(&payloadInfo, numEntries);
    extractPayloadInfo(schema, data, numEntries, schemaName, &payloadInfo);
    roctxRangePushA(payloadInfo.message);
```

Reply:
It will only allocate memory and extract payload info if the user flag is set "RCCL_LOG_ROCTX=1". By default this is set to 0.

Reply:
Is `roctxAlloc` not an API of the ROCTX library?

Reply:
No `roctxAlloc` and `roctxFree` are custom written for RCCL.